packages/plans-grid-next/src/components/features-grid/index.tsx

```diff
@@ -34,531 +33,777 @@ import PlanFeaturesContainer from '../plan-features-container';
+	translate,
+}: TabletViewProps ) => {
+	const gridPlansWithoutSpotlight = ! gridPlanForSpotlight
+		? gridPlans
+		: gridPlans.filter( ( { planSlug } ) => gridPlanForSpotlight.planSlug !== planSlug );
+	const numberOfPlansToShowOnTop = 4 === gridPlansWithoutSpotlight.length ? 2 : 3;
+	const plansForTopRow = gridPlansWithoutSpotlight.slice( 0, numberOfPlansToShowOnTop );
+	const plansForBottomRow = gridPlansWithoutSpotlight.slice( numberOfPlansToShowOnTop );
+	const tableProps = {
+		currentSitePlanSlug,
+		generatedWPComSubdomain,
+		gridPlanForSpotlight,
+		hideUnavailableFeatures,
+		intervalType,
+		isCustomDomainAllowedOnFreePlan,
+		isInSignup,
+		isLargeCurrency,
+		isLaunchPage,
+		onStorageAddOnClick,
+		onUpgradeClick,
+		paidDomainName,
+		planActionOverrides,
+		planUpgradeCreditsApplicable,
+		selectedFeature,
+		showUpgradeableStorage,
+		stickyRowOffset,
+		translate,
+	};
+
+	return (
+		<>
+			<div className="plan-features-2023-grid__table-top">
+				<Table gridPlans={ plansForTopRow } { ...tableProps } />
+			</div>
+			{ plansForBottomRow.length > 0 && (
+				<div className="plan-features-2023-grid__table-bottom">
+					<Table gridPlans={ plansForBottomRow } { ...tableProps } />
+				</div>
+			) }
+		</>
+	);
+};
 
-		return (
-			<>
-				{ this.renderSpotlightPlan() }
-				<div className="plan-features">
-					<div className="plan-features-2023-grid__content">
-						<div>
-							<div className="plan-features-2023-grid__desktop-view">
-								{ this.renderTable( gridPlans ) }
-							</div>
-							<div className="plan-features-2023-grid__tablet-view">
-								{ this.renderTabletView() }
-							</div>
-							<div className="plan-features-2023-grid__mobile-view">
-								{ this.renderMobileView() }
-							</div>
+const FeaturesGrid = ( {
+	gridPlans,
+	translate,
+	gridPlanForSpotlight,
+	stickyRowOffset,
+	isInSignup,
+	isLargeCurrency,
+	planUpgradeCreditsApplicable,
+	currentSitePlanSlug,
+	isLaunchPage,
+	planActionOverrides,
+	onUpgradeClick,
+	intervalType,
+	onStorageAddOnClick,
+	showUpgradeableStorage,
+	paidDomainName,
+	hideUnavailableFeatures,
+	selectedFeature,
+	generatedWPComSubdomain,
+	isCustomDomainAllowedOnFreePlan,
+}: FeaturesGridProps ) => {
+	const sharedProps = {
+		currentSitePlanSlug,
+		gridPlanForSpotlight,
+		intervalType,
+		isInSignup,
+		isLargeCurrency,
+		isLaunchPage,
+		onStorageAddOnClick,
+		onUpgradeClick,
+		planActionOverrides,
+		planUpgradeCreditsApplicable,
+		selectedFeature,
+		showUpgradeableStorage,
+		translate,
+	};
+
+	const additionalProps = {
+		generatedWPComSubdomain,
+		gridPlans,
+		hideUnavailableFeatures,
+		isCustomDomainAllowedOnFreePlan,
+		paidDomainName,
+	};
+
+	return (
+		<>
+			<SpotlightPlan { ...sharedProps } />
+			<div className="plan-features">
+				<div className="plan-features-2023-grid__content">
+					<div>
+						<div className="plan-features-2023-grid__desktop-view">
+							<Table
+								{ ...sharedProps }
+								{ ...additionalProps }
```

Comment:
Good idea, I think this can be simplified. I've updated (https://github.com/Automattic/wp-calypso/pull/86730/commits/a0350e61f49953679c4dc60198f6a47ab4dd4184) to use `spotlightPlanProps` and `planFeaturesProps`. I think the additional `planFeaturesProps` is helpful as it illustrates the commonality between `<Table />`, `<TabletView />` and `<MobileView />`.